reactivesocket/src/main/scala/akka/stream/alpakka/reactivesocket/scaladsl/HelloWorld.scala

```diff
@@ -0,0 +1,167 @@
+import io.reactivesocket.Payload
+import io.reactivesocket.client.ReactiveSocketClient
+import io.reactivesocket.transport.tcp.client.TcpTransportClient
+
+import io.reactivesocket.client.KeepAliveProvider._
+import io.reactivesocket.client.SetupProvider._
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Source
+import akka.stream.scaladsl.Sink
+import io.reactivesocket.util.PayloadImpl
+import scala.concurrent.Future
+import io.reactivesocket.frame.ByteBufferUtil
+import akka.stream.Materializer
+import akka.Done
+import akka.NotUsed
+
+object HelloWorld {
+  def main(args: Array[String]): Unit =
+    (new HelloWorld).start()
+}
+
+class HelloWorld {
+  val system = ActorSystem("HelloWorld")
+  implicit val materializer = ActorMaterializer.create(system)
+  import system.dispatcher
+
+  def start(): Unit = {
+
+    val serverHandler = new AbstractReactiveSocketApi {
+      override def requestResponse(payload: Payload): Future[Payload] = {
+        println(s"# got request") // FIXME
+        Future.successful(payload)
+      }
+    }
+    val server = ReactiveSocketServer.create(TcpTransportServer.create())
+      .start(ServerSocketAcceptorAdapter.disabledLease(setup => serverHandler))
+
+    val serverAddress = server.getServerAddress
+    println(s"started server: $serverAddress") // FIXME
+
+    val client: Future[ReactiveSocketApi] =
+      ClientReactiveSocketAdapter(ReactiveSocketClient.create(
+        TcpTransportClient.create(serverAddress),
+        keepAlive(never()).disableLease()
+      ))
+
+    client.foreach { socket =>
+      println(s"# got socket") // FIXME
+      val response: Future[Payload] = socket.requestResponse(new PayloadImpl("Hello"))
+
+      response.onFailure {
+        case e => println(e.getMessage)
+      }
+
+      response.map(_.getData)
+        .map(x => { println("got response"); x }) // FIXME remove
+        .map(ByteBufferUtil.toUtf8String)
+        .map(println)
+        .andThen { case _ => socket.close() }
+    }
+
+  }
+
+}
+
+trait ReactiveSocketApi {
+
+  def fireAndForget(payload: Payload): Future[Done]
+
+  def requestResponse(payload: Payload): Future[Payload]
+
+  def requestStream(payload: Payload): Source[Payload, NotUsed]
+
+  def requestChannel(payloads: Source[Payload, NotUsed]): Source[Payload, NotUsed]
+
+  def close(): Unit
+
+}
+
+trait AbstractReactiveSocketApi extends ReactiveSocketApi {
+
+  // FIXME real, empty implementations
+
+  def fireAndForget(payload: Payload): Future[Done] = ???
+
+  def requestResponse(payload: Payload): Future[Payload] = ???
+
+  def requestStream(payload: Payload): Source[Payload, NotUsed] = ???
+
+  def requestChannel(payloads: Source[Payload, NotUsed]): Source[Payload, NotUsed] = ???
+
+  def close(): Unit = ???
+
+}
+
+object ServerSocketAcceptorAdapter {
+
+  def disabledLease(handler: ConnectionSetupPayload => ReactiveSocketApi)(implicit materializer: Materializer): SocketAcceptor =
```

Comment:
I think it will be very common to use more than one interaction type in a microservice.
E.g. `sendTweet` (fire-n-forget), `getTweetDetails` (request-response), `getTweetStream` (request-stream).

If we dislike the `Abstract` pattern we could create a `Builder` dsl instead.

```scala
ReactiveSocketApi
  .withFireAndForget(...)
  .withRequestResponse(...)
```